asdf.elv

```diff
@@ -31,31 +31,31 @@ fn asdf [command @args]{
   }
 }
 
-fn match [argz @pats]{
+fn match {|argz @pats|
   var matched = $true;
   if (!= (count $argz) (count $pats)) {
-    matched = $false
+    set matched = $false
   } else {
     for i [(range (count $pats))] {
-      pat = '^'$pats[$i]'$'
-      arg = $argz[$i]
+      var pat = '^'$pats[$i]'$'
+      var arg = $argz[$i]
       if (not (re:match $pat $arg)) {
-        matched = $false
+        set matched = $false
         break
       }
     }
   }
   put $matched
 }
 
-fn ls-shims []{
+fn ls-shims {||
```

Comment:
```suggestion
fn ls-shims {
```

I'm aware the Elvish v0.17.0 upgrade script converts `[]{ body }` into `{|| body }` and I guess this is a matter of stylistic preference, but the `||` here isn't needed (the `[]` wasn't required in the legacy syntax as well). The [Elvish docs](https://elv.sh/ref/language.html#function) don't use `{|| body }` and instead use `{ body }`, and IMO `{ body }` looks cleaner.

Reply:
Yes, looks better without the `||`. Done